src/classes/BDI_DataImportBatch_TDTM.cls

```diff
@@ -0,0 +1,70 @@
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Batch Data Import
+* @group-content ../../ApexDocContent/BatchDataImport.htm
+* @description Trigger Handler for the DataImportBatch SObject
+*/
+public class BDI_DataImportBatch_TDTM extends TDTM_Runnable {
+    
+    public static final String BATCH_STATUS_OPEN = 'Open';
+    public static final String BATCH_STATUS_COMPLETED = 'Completed';
+    public static final String BATCH_STATUS_FAILED = 'Failed - Needs Review';
+
+    /*******************************************************************************************************************
+     * @description Trigger Handler on BatchDataImport
+     * @param newSObjects the list of records from trigger new.
+     * @param oldSObjects the list of records from trigger old.
+     * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+     * @param objResult the describe for BatchDataImport
+     * @return dmlWrapper.
+     ********************************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newSObjects, List<SObject> oldSObjects,
+        TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
+            handleBefore(newSObjects);
+        } 
+
+        return null;
+    }
+
+    /**
+     * @description In the before trigger context, sets the status for the batch
+     * @param List<DataImportBatch__c> 
+     */
+    private void handleBefore(List<DataImportBatch__c> batches) {
+        for (DataImportBatch__c b : batches) {
+            b.Batch_Status__c = BATCH_STATUS_OPEN;
+        }
```

Comment:
Just curious here - could this field be given a default value instead of using a trigger to always set new records to a single value?

Reply:
Good point, and I did consider it. I looked at what we do for the di status picklist, and we do not set a default value in that case, that's why I went with the trigger for the batch. I am open to go with the default value though, but can we quickly chat tomorrow? Let's also talk about the labels.